scripts/svgGraphs.js

```diff
@@ -70,6 +70,65 @@ const {barLineGraph, stackedAreaGraph, pieGraph} = (function () {
         }
     }
 
+    const nodeIterator = (elem, fn) => {
+        fn(elem);
+        for (const child of elem.children) {
+            nodeIterator(child, fn);
+        }
+    };
+
+    const createSvgText = (svgElement) => {
+        const classNames = new Set(['graph-background']);
+
+        nodeIterator(svgElement, (elem) => {
+            elem.classList.forEach(className => {
+                classNames.add(className);
+            });
+        });
+
+        const dummyElement = document.createElement('a');
+        classNames.forEach(className => {
+            dummyElement.classList.add(className);
+        });
+        const cssStyleRules = new Set();
+        for (const sheet of document.styleSheets) {
+            for (const rule of sheet.cssRules) {
+                if (dummyElement.matches(rule.selectorText)) {
+                    cssStyleRules.add(rule.cssText);
+                }
+            }
+        }
+
+        let svgText = svgElement.outerHTML.replace('<svg ', '<svg xmlns="http://www.w3.org/2000/svg" width="450px" ');
```

Comment:
I think users are going to expect the downloaded SVG to be the same size as the one they had on the screen, not smaller. I know this isn't Linda's use-case but this is now a button for everyone for (almost) every graph.

Is it possible to set the downloaded width to the _current_ width of the on-page SVG? Then Linda could narrow her screen size to the width she wants and just download them then.

Reply:
Good idea. I've implemented this.